packages/api/src/background/FinchPortManager.ts

```diff
@@ -0,0 +1,145 @@
+import {
+  onConnectExternal,
+  onConnect,
+  removeConnectExternalListener,
+  removeConnectListener,
+} from '@finch-graphql/browser-polyfill';
+import { FinchDefaultPortName } from '@finch-graphql/types';
+import { FinchMessage } from '..';
+import { FinchDevtoolsIncomingMessage } from './types';
+
+type AnyMessage = (FinchDevtoolsIncomingMessage | FinchMessage<unknown>) & {
+  id: string;
+};
+
+interface FinchPortManagerOptions {
+  onDevtoolMessage: (msg: AnyMessage) => Promise<object>;
+  onMessage: (msg: AnyMessage) => Promise<object>;
+  messagePortName?: string;
+  external?: boolean;
+}
+
+export class FinchPortManager {
```

Comment:
Let's move this to a general port manager and take out the idea of this being a dev-port or a message port. That way we can create two of these, and make it explicitly for dev tools, and the other for messaging.